app/frontend/wallet/shelley/shelley-transaction-planner.ts

```diff
@@ -313,7 +314,11 @@ const validateTxPlan = (txPlan: TxPlan): void => {
     throw NamedError('OutputTooSmall')
   }
   const totalRewards = txPlan.withdrawals.reduce((acc, {rewards}) => acc + rewards, 0)
-  if (totalRewards > 0 && totalRewards < txPlan.fee) {
+  const deregisteringStakeKey = txPlan.certificates.some(
+    (c) => c.type === CertificateType.STAKING_KEY_DEREGISTRATION
+  )
+  // TODO: confirm this "deregisteringStakeKey" condition
+  if (!deregisteringStakeKey && totalRewards > 0 && totalRewards < txPlan.fee) {
```

Comment:
Please check if this condition makes sense.
The idea here is that I assume that when deregistering the "stakeKey" the returned "deposit" should be always greater than the "fee" even if the "rewards" are non-zero and less than "fee". Is this assumption correct? Maybe also question on @refi93 ?